### PR TITLE
Remove remaining `# pylint` in `client-superstaq`

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Removed any remaining `# pylint` from non-test files throughout `client-superstaq` after merging main (i.e., completion of https://github.com/Infleqtion/client-superstaq/pull/503 and https://github.com/Infleqtion/client-superstaq/pull/513 https://github.com/Infleqtion/client-superstaq/pull/578).

### CODE REVIEW

**Review by vtomole — COMMENTED** [review]
We want the checks to [have docstrings](https://github.com/Infleqtion/client-superstaq/pull/503) so we shouldn't ignore those files

**Comment by vtomole** [comment]
The `qubit_gates` does are here https://github.com/Infleqtion/client-superstaq/pull/503. You can merge that branch into this one.

**Comment by bharat-thotakura** [comment]
^ Ah noted, will update accordingly.

**Review by vtomole — COMMENTED** [review]
Let's merge https://github.com/Infleqtion/client-superstaq/pull/503 into this PR as well. The plan would be to merge said PR and https://github.com/Infleqtion/client-superstaq/pull/513 before merging in this one.

**Comment by bharat-thotakura** [comment]
Sure, I did merge #503, but will wait for both #503 and #513 to be merged in first.

**Comment by vtomole** [comment]
Let's pull out the easy part of this PR (adding `# pylint: disable=missing-function-docstring` at the top of test files) into a new PR. That will be an easy review/merge then we can address the docstrings in this PR, #563 and #513

**Comment by bharat-thotakura** [comment]
^ @vtomole, see #578 per your suggestion

**Comment by pgoiporia** [comment]
I think my PR might have caused the merge conflicts on this. Will resolve them and push

File: cirq-superstaq/cirq_superstaq/compiler_output_test.py
Comment (vtomole) [inline cirq-superstaq/cirq_superstaq/compiler_output_test.py:232]: Did you add this? Only asking cause it's not related to the PR title.

Comment (bharat-thotakura) [reply cirq-superstaq/cirq_superstaq/compiler_output_test.py:232]: Yes, removing ` # pylint: disable=missing-param-doc` in `compiler_output.py` failed the coverage test so I added that test.

File: cirq-superstaq/cirq_superstaq/ops/qubit_gates.py
Comment (vtomole) [inline cirq-superstaq/cirq_superstaq/ops/qubit_gates.py:600]: I know we are inconsistent with docstrings but https://google.github.io/styleguide/pyguide.html#384-classes suggests we write a one sentence summary after the `__init__`

```python
def __init__(self, likes_spam: bool = False):
        """Initializes the instance based on spam preference.

        Args:
          likes_spam: Defines if instance exhibits this preference.
        """
        self.likes_spam = likes_spam
        self.eggs = 0
```

Comment (bharat-thotakura) [reply cirq-superstaq/cirq_superstaq/ops/qubit_gates.py:600]: Yes, see https://github.com/Infleqtion/client-superstaq/pull/585

**Comment by bharat-thotakura** [comment]
@vtomole I removed the newline after `# pylint: disable=missing-function-docstring,missing-class-docstring` if that's  preferred. Otherwise, will just revert [7a73afa](https://github.com/Infleqtion/client-superstaq/pull/565/commits/7a73afa6582e3654f9e77fdf9e276c2fbbdc4e66).

**Comment by vtomole** [comment]
I could go either way. No need to revert.

File: cirq-superstaq/cirq_superstaq/compiler_output_test.py
Comment (vtomole) [reply cirq-superstaq/cirq_superstaq/compiler_output_test.py:231]: I commented out this code in https://github.com/Infleqtion/client-superstaq/pull/586 and the checks pass.

Comment (bharat-thotakura) [reply cirq-superstaq/cirq_superstaq/compiler_output_test.py]: Yeah saw the test PR too, removed the tests with [6d115a3](https://github.com/Infleqtion/client-superstaq/pull/565/commits/6d115a3bf55825703943d785e30f050482397e39) (not sure why I saw lines 231-239 in `cirq_superstaq/compiler_ouput.py` flagged during one of my earlier `./check/all_.py -i` checks locally)

File: cirq-superstaq/cirq_superstaq/compiler_output.py
Comment (pgoiporia) [inline cirq-superstaq/cirq_superstaq/compiler_output.py:167]: There's probably enough arguments here to warrant an arguments and return section

Comment (vtomole) [reply cirq-superstaq/cirq_superstaq/compiler_output.py:167]: These will be fixed in a different PR.

**Review by vtomole — APPROVED** [review]
LGTM

File: cirq-superstaq/cirq_superstaq/compiler_output.py
Comment (bharat-thotakura) [reply cirq-superstaq/cirq_superstaq/compiler_output.py:167]: Are you referring to more text for the current [docstring](https://github.com/Infleqtion/client-superstaq/blob/442c8a7e095961544cd2bd1fa23e1c0d75f77b06/cirq-superstaq/cirq_superstaq/compiler_output.py#L167-L178)?

Comment (pgoiporia) [reply cirq-superstaq/cirq_superstaq/compiler_output.py:167]: @bharat-thotakura my bad, looks like I was looking at the wrong version of the file! docstring looks good.

**Pull request merged**
